tests/python_client/testcases/test_utility.py

```diff
@@ -2309,3 +2390,1259 @@ def test_role_list_grants(self, host, port):
+        u, _ = self.utility_wrap.create_user(user=user, password=password)
+
+        self.utility_wrap.init_role(name)
+        self.utility_wrap.role_add_user(user)
+        users, _ = self.utility_wrap.role_get_users()
+        user_info, _ = self.utility_wrap.list_user(user, True)
+        user_item = user_info.groups[0]
+        assert name in user_item.roles
+        assert user in users
+
+        self.utility_wrap.role_remove_user(user)
+        users, _ = self.utility_wrap.role_get_users()
+        assert user not in users
+
+    @pytest.mark.tags(CaseLabel.L3)
+    def test_remove_root_from_new_role(self, host, port):
+        """
+        target: remove root from new role
+        method: create a new role, bind root to role,remove root from role
+        expected: remove success
+        """
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+
+        r_name = cf.gen_unique_str(prefix)
+        self.utility_wrap.init_role(r_name)
+        self.utility_wrap.create_role()
+        assert self.utility_wrap.role_is_exist()[0]
+        self.utility_wrap.role_add_user("root")
+        users, _ = self.utility_wrap.role_get_users()
+        user_info, _ = self.utility_wrap.list_user("root", True)
+        user_item = user_info.groups[0]
+        assert r_name in user_item.roles
+        assert "root" in users
+
+        self.utility_wrap.role_remove_user("root")
+        users, _ = self.utility_wrap.role_get_users()
+        assert "root" not in users
+        self.utility_wrap.role_drop()
+
+
+class TestUtilityNegativeRbac(TestcaseBase):
+    @pytest.mark.tags(CaseLabel.L3)
+    @pytest.mark.parametrize("name", ["longlonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong"
+                                      "longlonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong"
+                                      "longlonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong"
+                                      "longlonglonglong",
+                                      "%$#@!", "123", " ", "''", "test-role"])
+    def test_create_role_with_invalid_name(self, name, host, port):
+        """
+        target: create role with invalid name
+        method: create role with invalid name
+        expected: create fail
+        """
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+        self.utility_wrap.init_role(name)
+
+        error = {"err_code": 5}
+        self.utility_wrap.create_role(check_task=CheckTasks.err_res, check_items=error)
+        # get roles
+        role_groups, _ = self.utility_wrap.list_roles(False)
+
+        # drop roles
+        for role_group in role_groups.groups:
```

Comment:
what does this loop try to verify?

Reply:
init for negative test case.max roles number limit.